fendermint/app/src/app.rs

```diff
@@ -650,7 +684,13 @@ where
             .context("failed to prepare proposal")?;
 
         let txs = txs.into_iter().map(bytes::Bytes::from).collect();
-        let txs = take_until_max_size(txs, request.max_tx_bytes.try_into().unwrap());
+        let (txs, size) = take_until_max_size(txs, request.max_tx_bytes.try_into().unwrap());
+
+        emit(BlockProposalSent {
+            height: request.height.value(),
+            tx_count: txs.len(),
```

Comment:
It would be useful to print our own validator address here, which is part of `request::PrepareProposal`.

Reply:
Yes, good point. Added it.